### DISCUSSION TITLE
Email editor modification

### DISCUSSION THREAD

**Discussion by aioun** [linagora/tmail-flutter#1334] [Ideas]
Hello,
I want to change a little bit the email editor widget by adding a new feature (example in the image below). However, I don't want to study all tmail code, If you can help me with which class or file I should change it. 
Thanks
![Capture d’écran de 2022-12-19 16-05-22](https://user-images.githubusercontent.com/42891685/208457825-65412aad-d69e-42dd-803b-44aeceb77537.png)

**Comment by hoangdat** [comment]
Hi, TeamMail is separate by functions, you can find the `composer` in `features` directory.

<img width="315" alt="image" src="https://user-images.githubusercontent.com/6462404/208538574-3ea76dd3-f778-4df0-9230-08bdb2b46cc7.png">

You can add your modification here.

`Note:` 
- `composer_view.dart`: use for mobile/tablet screen
- `composer_view_web.dart`: use for browser screen (responsive)

**Comment by aioun** [comment]
Thank you very much. I was suspecting these files but I wanted to be sure.
I developed an external example using flutter for test. It works and now I will proceed to integration.

**Reply by aioun** [reply]
Done. I can work around my email quickly. However, I can't use the TAB keyboard.

![output](https://user-images.githubusercontent.com/42891685/209323628-1bc23c41-047c-445c-a1f5-3ef85983ab44.gif)

**Reply by hoangdat** [reply]
oh, so cool, this is smart suggestion for editor? Can you create a PR?

**Comment by aioun** [comment]
Thanks, it is not so easy to make PR before I explain the functionning ! 
I am using a heavy AI model, about 200 Mo. From my understanding, there are two solutions to integrate the model:
1- put the model in assets folder , but in this case  client has to wait for loading html , images, model file and etc.
2- serve the model
I am using the second approach. I was testing with a websocket server, it works fine but does not fit a great number of clients, using this feature at the same time. Thus, I developed a phonix-elixir server using poolboy and phonix channels to handle great number of concurrent users.
Besides, in the feature, the goal is to personalize the model to each user. 
That's it !
